bluemira/display/auto_config.py

```diff
@@ -48,6 +50,27 @@ def get_primary_screen_size():
     height: Union[int, None]
         height of the primary screen in pixels. If there is no screen returns None
     """
+    pool = Pool(processes=1)
+    result = pool.apply_async(_get_primary_screen_size)
+    try:
+        val = result.get(timeout=3)
+    except TimeoutError:
+        pool.terminate()
+        bluemira_warn(
+            "Unable to get screensize, please check your X server."
+            " You may not be able to view live figures in this mode."
+        )
+        return None, None
+    else:
+        pool.close()
+        pool.join()
+        return val
+
+
+def _get_primary_screen_size():
+    """
+    Direct run of screen size check without subprocess
+    """
     if sys.platform.startswith("linux") and os.getenv("DISPLAY") is None:
         return None, None
```

Comment:
Should we also be warning the user in this case?

Reply:
Hmm, I sort of think if you don't have a DISPLAY set up in any way it probably means that you don't have or want anything visual to pop up. Maybe a debug message would be enough?